### DISCUSSION TITLE
RPI Camera v2 NoIR - applying the imx219_noir tuning file with motionplus?

### DISCUSSION THREAD

**Discussion by hibit74** [Motion-Project/motionplus#130] [General]
Hello all - I wonder if anyone can point me to how I would apply the tuning file when using motionplus?

I have motionplus running nicely here with the camera configured as a libcam device in the motionplus config file - but with the red hue to the image. The camera is detected as a IMX219 in list-cameras as I understand there is no hw difference between the standard and noir versions of the v2 cam.

Pre-libcam I was able to set awb to isgreyworld, and more recently using libcamera alone (without motionplus) I can set the config file when calling libcamera on the command line.

Is there a way to configure the tuning file to be used within libcam_params in the motionplus config that I am missing?
Or can I specify the tuning file at boot before motionplus is run?

Appreciate any pointers, many thanks

**Comment by Mr-Dave** [comment]
Yes, you can set "most" of the libcamera options via the `libcam_params` option in Motionplus.  The options available to be set are in the [Motionplus documentation](https://motion-project.github.io/motionplus_config.html#libcam_params).  For example, to set the AWB to custom, the Motionplus parameter would be specified like the following:

`libcam_params AwbMode=7`

As for which parameters to set to what, that is a libcamera thing.  This is the [libcamera api documentation reference](https://libcamera.org/api-html/namespacelibcamera_1_1controls.html).  I didn't see  "isgreyworld" so you'd need to experiment to find the equivalent.

Additional Note:  This discussion was moved from the Motion repository since libcam_params is only applicable to Motionplus.

**Reply by hibit74** [reply]
Thanks for getting back to me Mr Dave and for moving this to right repo. To add a little more background.
From the command line if I execute motionplus normally as a service with my motionplus config set to expect a libcamera device it runs fine, jsut produces the red-hued output. However, if I execute this at the command line:

libcamerify --tuning-file /usr/share/libcamera/ipa/raspberrypi/imx219_noir.json motionplus

This runs but produces a black video stream as libcamerify (I think) presents a v4l device to motionplus, which my config doesn't expect (it is set up for libcamera device). To note, running the likes of:

libcamera-hello --tuning-file /usr/share/libcamera/ipa/raspberrypi/imx219_noir.json motionplus

Works fine and applies the tuning file, fixing the red hue.

What I can't find a way to do, is run motionplus whilst also applying the tuning file. Should I set me motionplus config to expect a v4l device, use libcamerify to 'wrap' motionplus at the command line? I tried this but couldn't seem to make that work.

**Reply by Mr-Dave** [reply]
You'll need to change whatever is inside the json file into parameters for the libcam_params
